### PR TITLE
Fix crash when assigning more textures than expected to texture array

### LINKED ISSUE #93144: memory overflow when shader's uniform array out of range
### Tested versions

4.2.2 stable

### System information

windows 10

### Issue description

yesterday I wrote a shader with a uniform Array:
```uniform sampler2D[7] gradient :source_color```
but I forgot the uplimit range.Then I added 11 gradient color in inspector.
Then memory overflow and Godot editor crashed. I can't open my project anymore until I understood the reason.
I think the inspector should stop me add child when the child count over the up limit of array.
Or the engine should just give me a helpful error instead of crashing

### Steps to reproduce

1. download this MRP and open this project if you can.
2. open control.tscn. It is a simple control with a subview container and a label.
3. the subview container have a materail.Open the inspector you can see there is 5 gradient color have been defined.
4. open the popupfont.gdshader file. you can see sampler2D[2]. So it will memory overflow.
5. run the game and press A. the script control.gd will do this:
```
var obj = SUB_VIEWPORT_CONTAINER.instantiate() as SubViewportContainer
obj.material = obj.material.duplicate()
```
6. Then you will get the Bug: first our game crashed without any error massage. 
7. At this time, maybe we want to try restart godot editor. So, close the editor.
8. try to open this project again. You will get the editor crashing.
9. delete the .godot folder or delete the control.tscn you can open project but it will crash again just a while

### Minimal reproduction project (MRP)

[test.zip](https://github.com/user-attachments/files/15830473/test.zip)

### PR DESCRIPTION
**Pull request by Chaosus**
- Fix https://github.com/godotengine/godot/issues/93144

### CODE REVIEW

File: servers/rendering/renderer_rd/storage_rd/material_storage.cpp [hunk 1]
```diff
@@ -807,8 +807,10 @@ void MaterialStorage::MaterialData::update_textures(const HashMap<StringName, Va
 				if (V->value.is_array()) {
 					Array array = (Array)V->value;
 					if (uniform_array_size > 0) {
-						for (int j = 0; j < array.size(); j++) {
-							textures.push_back(array[j]);
+						for (int j = 0; j < uniform_array_size; j++) {
```
Comment (AThousandShips) [inline servers/rendering/renderer_rd/storage_rd/material_storage.cpp:810]:
```suggestion
						for (int j = 0; j < MIN(uniform_array_size, array.size()); j++) {
```
Instead, and remove the check

File: drivers/gles3/storage/material_storage.cpp [hunk 2]
```diff
@@ -874,8 +874,10 @@ void MaterialData::update_textures(const HashMap<StringName, Variant> &p_paramet
 				if (V->value.is_array()) {
 					Array array = (Array)V->value;
 					if (uniform_array_size > 0) {
-						for (int j = 0; j < array.size(); j++) {
-							textures.push_back(array[j]);
+						for (int j = 0; j < uniform_array_size; j++) {
```
Comment (AThousandShips) [inline drivers/gles3/storage/material_storage.cpp:877]: Same here

**Review by AThousandShips — APPROVED** [review]
Makes sense

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**